### PR TITLE
React Dapp Tutorial

### PR DESCRIPTION
**Pull request by httpJunkie**
This is part 1 of a 2-part series. For now, this tutorial will stand on its own. 

This PR is connected to issue #709

### CODE REVIEW

**Comment by httpJunkie** [comment]
Can we get this PR reviewed, we have another part coming as soon as we can get this one merged. Or I can drop this PR and create a new one.

File: wallet/tutorials/simple-react-dapp.md [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+cd mm-dapp-react && npm install
+```
+
+The `npx vite` or `npm run dev` commands can be used to run our project at any time if the development server has been stopped.
+
+Open the project in your code editor of choice. We will reset the `App.tsx` page to give us a blank slate.
+
+Update `App.tsx` to:
+
+```ts
```
Comment (ziad-saab) [inline wallet/tutorials/simple-react-dapp.md:62]: use `tsx` instead, you get better syntax highlight

File: wallet/tutorials/simple-react-dapp.md [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+
+Install the dependency:
+
+```bash
+npm install @metamask/detect-provider
+```
+
+With this, we can update our `src/App.tsx` to:
+
+```ts
```
Comment (ziad-saab) [inline wallet/tutorials/simple-react-dapp.md:162]: use `tsx` instead, you get better syntax highlight

File: wallet/tutorials/simple-react-dapp.md [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+
+### React We Have a Problem
+
+We lose our account data if we refresh the page. When we connect with the button, we set `accounts` in our state, but when the page refreshes, we need something in our `useEffect` to check if we already have connected and update our wallet state.
+
+Thinking ahead, we know that once we are tracking more than just `accounts`, we will also need a mechanism to get the `balance` and `chainId` and update their state.
+
+Let's update our `src/App.tsx` with some added logic to our `useEffect`:
+
+```ts
```
Comment (ziad-saab) [inline wallet/tutorials/simple-react-dapp.md:283]: use `tsx` instead, you get better syntax highlight

File: wallet/tutorials/simple-react-dapp.md [hunk 4]
```diff
@@ -0,0 +1,625 @@
...
+
+With those functions exported we will be able to import them into our component and use them as functions where needed to get human-readable balance and chain information.
+
+### Watch User Balance and Chain
+
+To update our current component for displaying the connected address's balance and the current `chainId`, we need to update our `initialState` object. Since we already use the `eth_requestAccounts` RPC endpoint to determine the accounts, we need to add a dependent call to `eth_getBalance` once we have that account information.
+
+Finally, we need to parse the returned value of the balance and format it into a human-readable string. We'll create a function called `formatBalance` as well.
+
+```ts
```
Comment (ziad-saab) [inline wallet/tutorials/simple-react-dapp.md:387]: use `tsx` instead, you get better syntax highlight

**Review by ziad-saab — APPROVED** [review]

**Comment by alexandratran** [comment]
Working on editing this today – will integrate Ziad's suggestions.

**Review by alexandratran — APPROVED** [review]
Edited the content of the tutorial for style and simplicity. Lmk if you have any questions about the edits. The PR looks good to me!

**Pull request merged**